### PR TITLE
Function to convert new SymbolProblems to old LinkageCheckReport

### LINKED ISSUE #622: A utility function to convert new value class (SymbolProblem) to old value class (LinkageCheckReport))
Part of the refactoring (#574), especially #600. 

```
  public LinkageCheckReport findLinkageErrors() {
  ...
```
is going to change to new value classes
```
  public ImmutableSetMultimap<ClassFile, SymbolProblem> findSymbolProblems() {
  ...
```

The utility function convert the new value classes back to old one. The caller (dashboard and enforcer rule) uses the function.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #622 . This is a preparation for #600 .

### CODE REVIEW

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReport.java [hunk 1]
```diff
@@ -48,4 +60,98 @@ public String getErrorString() {
...
+      case INACCESSIBLE_MEMBER:
+        return SymbolNotResolvable.errorInaccessibleMember(
+            symbolReference, targetClassLocation, isReachable);
+      case SYMBOL_NOT_FOUND:
+        return SymbolNotResolvable.errorMissingMember(
+            symbolReference, targetClassLocation, isReachable);
+      case CLASS_NOT_FOUND:
+        return SymbolNotResolvable.errorMissingTargetClass(symbolReference, isReachable);
+      default:
+        throw new UnsupportedOperationException(
```
Comment (elharo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReport.java]: I'm not sure this is the right exception here, though I can't put my finger on one that fits. What if you moved this method into the SymbolNotResolvable type instead?

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReport.java]: Moved the function to SymbolNotResolvable and updated the exception to VerifyException (as per [ConditionalFailuresExplained](https://github.com/google/guava/wiki/ConditionalFailuresExplained#summary); "Someone I depend on messed up.")

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java [hunk 2]
```diff
@@ -176,4 +177,31 @@ public String toString() {
      */
     SYMBOL_NOT_FOUND
   }
+
+  static <U extends SymbolReference> SymbolNotResolvable<U> fromSymbolProblem(
```
Comment (elharo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java]: Can this be an instance method on SymbolProblem so you don't need the switch at all?

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java]: The switch statement will remain even if I move the method to an instance method on SymbolProblem.
 (Or I’m missing something?)

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java [hunk 3]
```diff
@@ -143,37 +143,15 @@ public String toString() {
     abstract SymbolNotResolvable<T> build();
   }
 
-  /** The kind of linkage error against a symbol reference. */
-  enum Reason {
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java]: This enum (old) was duplicate of `ErrorType` (new). They have the same set of constants. The old enum was going to be removed in the course of the refactoring anyway, but I found changing this to `ErrorType` now helps to delete the switch statement.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java (hunk 2 above)
Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolNotResolvable.java]: Now the function has no switch statement.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReportTest.java [hunk 4]
```diff
@@ -111,4 +118,52 @@ public void testCreation() {
             .getReference()
             .getTargetClassName());
   }
+
+  @Test
+  public void testConversion() throws IOException, URISyntaxException {
+    Path path1 = absolutePathOfResource("testdata/grpc-google-cloud-firestore-v1beta1-0.28.0.jar");
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReportTest.java]: grpcPath would be more descriptive than path1

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReportTest.java [hunk 5]
```diff
@@ -111,4 +118,52 @@ public void testCreation() {
...
+    Path path2 = absolutePathOfResource("testdata/google-cloud-firestore-0.65.0-beta.jar");
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReportTest.java]: similarly, firestorePath

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReportTest.java [hunk 6]
```diff
@@ -111,4 +118,52 @@ public void testCreation() {
...
+            new ClassFile(path1, "com.google.firestore.v1beta1.FirestoreGrpc"), symbolProblem1,
+            new ClassFile(path2, "com.google.ClassB"), symbolProblem2);
+
+    SymbolReferenceMaps.Builder referenceMapBuilder = new SymbolReferenceMaps.Builder();
+    LinkageCheckReport linkageCheckReport =
+        LinkageCheckReport.fromSymbolProblems(
+            symbolProblems,
+            ImmutableList.of(path1, path2),
+            ClassReferenceGraph.create(referenceMapBuilder.build(), ImmutableSet.of(path1)));
+    Truth.assertThat(linkageCheckReport.getJarLinkageReports()).hasSize(2);
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReportTest.java]: Truth doesn't seem to add anything compared to assertEquals in this method, and bloats the code a bit.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerTest.java [hunk 7]
```diff
@@ -153,7 +152,7 @@ public void testCheckLinkageErrorMissingInterfaceMethodAt_interfaceAndClassSepar
         linkageChecker.checkLinkageErrorMissingMethodAt(methodSymbolReference);
 
     Truth8.assertThat(errorFound).isPresent();
-    Truth.assertThat(errorFound.get().getReason()).isEqualTo(Reason.INCOMPATIBLE_CLASS_CHANGE);
+    Truth.assertThat(errorFound.get().getReason()).isEqualTo(ErrorType.INCOMPATIBLE_CLASS_CHANGE);
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerTest.java]: Truth doesn't seem to add anything compared to assertEquals in this method, and bloats the code a bit.

**Pull request merged**